src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -218,6 +232,10 @@ public void InitializeService(ServiceHost serviceHost)
                 {
                     await requestContext.SendError(SR.QueryServiceResultSetTooLarge);
                     ActiveQueries.TryRemove(executeStringParams.OwnerUri, out removedQuery);
+                    ConnectionService.Disconnect(new DisconnectParams(){
```

Comment:
Why not make this a finally or something instead? You always want to disconnect, so let's ensure in the code that it happens no matter what the result of the query.